include/aos/common/pkcs11/pkcs11.hpp

```diff
@@ -0,0 +1,653 @@
+
+        return {};
+    }
+
+    /**
+     * Initializes a token.
+     *
+     * @param slotID slot identifier.
+     * @param pin user pin.
+     * @param label token label.
+     * @return Error.
+     */
+    Error InitToken(uint32_t slotID, const String& pin, const String& label)
+    {
+        (void)slotID;
+        (void)pin;
+        (void)label;
+
+        return {};
+    }
+
+    /**
+     * Returns list of slots in the system.
+     *
+     * @param tokenPresent
+     * @param[out] slotList result slot list.
+     * @return Error.
+     */
+    Error GetSlotList(bool tokenPresent, Array<uint32_t>& slotList) const
+    {
+        (void)tokenPresent;
+        (void)slotList;
+
+        return {};
+    }
+
+    /**
+     * Returns information about slot.
+     *
+     * @param slotID slot id.
+     * @param[out] slotInfo result slot information.
+     * @return Error.
+     */
+    Error GetSlotInfo(uint32_t slotID, SlotInfo& slotInfo) const
+    {
+        (void)slotID;
+        (void)slotInfo;
+
+        return {};
+    }
+
+    /**
+     * Returns token information for the specified slot id.
+     *
+     * @param slotID slot id.
+     * @param[out] tokenInfo token information for the slot.
+     * @return Error.
+     */
+    Error GetTokenInfo(uint32_t slotID, TokenInfo& tokenInfo) const
+    {
+        (void)slotID;
+        (void)tokenInfo;
+
+        return {};
+    }
+
+    /**
+     * Returns general information about Cryptoki library.
+     *
+     * @param[out] libInfo result information.
+     * @return Error.
+     */
+    Error GetLibInfo(LibInfo& libInfo) const
+    {
+        (void)libInfo;
+
+        return {};
+    }
+
+    /**
+     * Opens a session between an application and a token in a particular slot.
+     *
+     * @param slotID slot identifier.
+     * @param flags  indicates the type of session.
+     * @return Error.
+     */
+    RetWithError<UniquePtr<SessionContext>> OpenSession(uint32_t slotID, uint32_t flags)
+    {
+        (void)slotID;
+        (void)flags;
+
+        return {nullptr, Error::Enum::eNone};
+    }
+
+    /**
+     * Destroys object instace.
+     */
+    virtual ~LibraryContext() = default;
+};
+
+/**
+ * A Curve represents a short-form Weierstrass curve.
+ */
+enum class EllipticCurve {
+    /**
+     * A Curve which implements NIST P-384 (FIPS 186-3, section D.2.4), also known as secp384r1
+     **/
+    eP384
+};
+
+/**
+ * Crypto private key extended with object handle.
+ */
+struct PrivateKey {
+    /**
+     * Pointer to crypto private key.
+     */
+    SharedPtr<crypto::PrivateKey> mKey;
+    /**
+     * pkcs11 object handle.
+     */
+    ObjectHandle mHandle;
+};
+
+/**
+ * A helper class for a base Cryptoki API.
+ */
+class Utils {
+public:
+    /**
+     * Creates an object instance.
+     *
+     * @param session session context.
+     * @param allocator allocator for token/session objects.
+     */
+    Utils(pkcs11::SessionContext& session, Allocator& allocator)
```

Comment:
Sorry, I didn't find the example of `Utils` usage. I'm trying to understand allocator usage. Let's discuss again.